feature-libs/cart/main/core/facade/selective-cart.service.ts

```diff
@@ -1,199 +1,134 @@
-  protected userId: string;
-  protected cartId: string;
   protected selectiveCart$: Observable<Cart>;
-  protected cartId$: BehaviorSubject<string> = new BehaviorSubject<string>('');
-
-  protected readonly PREVIOUS_USER_ID_INITIAL_VALUE =
-    'PREVIOUS_USER_ID_INITIAL_VALUE';
-  protected previousUserId = this.PREVIOUS_USER_ID_INITIAL_VALUE;
-
-  protected cartSelector$ = this.cartId$.pipe(
-    switchMap((cartId) => {
-      this.cartId = cartId;
-      return this.multiCartService.getCartEntity(cartId);
-    })
-  );
 
   constructor(
-    protected store: Store<StateWithMultiCart>,
     protected userService: UserService,
     protected multiCartService: MultiCartService,
     protected baseSiteService: BaseSiteService,
     protected userIdService: UserIdService
-  ) {
-    combineLatest([
-      this.userService.get(),
-      this.baseSiteService.getActive(),
-    ]).subscribe(([user, activeBaseSite]) => {
-      if (user && user.customerId && activeBaseSite) {
-        this.customerId = user.customerId;
-        this.cartId$.next(`selectivecart${activeBaseSite}${this.customerId}`);
-      } else if (user && !user.customerId) {
-        this.cartId$.next('');
-      }
-    });
-
-    this.userIdService.getUserId().subscribe((userId) => {
-      this.userId = userId;
-
-      if (this.isJustLoggedIn(userId)) {
-        this.load();
-      }
-
-      this.previousUserId = userId;
-    });
-
-    this.selectiveCart$ = this.cartSelector$.pipe(
-      map(
-        (
-          cartEntity: StateUtils.LoaderState<Cart>
-        ): {
-          cart: Cart;
-          loading: boolean;
-          loaded: boolean;
-        } => {
-          return {
-            cart: cartEntity.value as Cart,
-            loading: Boolean(cartEntity.loading),
-            loaded: Boolean(
-              (cartEntity.error || cartEntity.success) && !cartEntity.loading
-            ),
-          };
-        }
-      ),
-      filter(({ loading }) => !loading),
-      tap(({ cart, loaded }) => {
-        if (this.cartId && this.isEmpty(cart) && !loaded) {
-          this.load();
-        }
-      }),
-      map(({ cart }) => (cart ? cart : {})),
-      shareReplay({ bufferSize: 1, refCount: true })
-    );
-  }
+  ) {}
 
+  /**
+   * Initialize the stream when first call this function
+   */
   getCart(): Observable<Cart> {
+    if (!this.selectiveCart$) {
+      this.selectiveCart$ = combineLatest([
+        this.getSelectiveCartId(),
+        this.userService.get(),
+        this.userIdService.getUserId(),
+        this.baseSiteService.getActive(),
+      ]).pipe(
+        distinctUntilChanged(),
+        tap(([selectiveId, user, userId, activeBaseSite]) => {
+          if (
+            !Boolean(selectiveId) &&
+            userId !== OCC_USER_ID_ANONYMOUS &&
+            user?.customerId
+          ) {
+            this.multiCartService.loadCart({
+              userId: userId,
+              cartId: `selectivecart${activeBaseSite}${user.customerId}`,
```

Comment:
iid `selectivecart${activeBaseSite}${user.customerId}` the same as `selectiveId` ?